docs/sources/flow/reference/components/prometheus.exporter.snmp.md

```diff
@@ -0,0 +1,169 @@
+---
+# NOTE(rfratto): the title below has zero-width spaces injected into it to
+# prevent it from overflowing the sidebar on the rendered site. Be careful when
+# modifying this section to retain the spaces.
+#
+# Ideally, in the future, we can fix the overflow issue with css rather than
+# injecting special characters.
+
+title: prometheus.exporter.snmp
+---
+
+# prometheus.exporter.snmp
+The `prometheus.exporter.snmp` component embeds
+[`snmp_exporter`](https://github.com/prometheus/snmp_exporter). `snmp_exporter` lets you collect SNMP data and expose them as Prometheus metrics.
+
+## Usage
+
+```river
+prometheus.exporter.snmp "LABEL" {
+  config_file = "PATH_SNMP_CONFIG_FILE"
+  
+  target "TARGET_NAME" {
+    address = "TARGET_ADDRESS" 
+  }
+
+  walk_param "PARAM_NAME" {
+  }
+  
+  ...
+}
+```
+
+## Arguments
+The following arguments can be used to configure the exporter's behavior.
+Omitted fields take their default values.
+
+Name | Type | Description | Default | Required
+---- | ---- | ----------- | ------- | --------
+`config_file` | `string`       | SNMP configuration file with custom modules. | | yes
+
+The `config_file` argument points to a YAML file defining which snmp_exporter modules to use. See [snmp_exporter](https://github.com/prometheus/snmp_exporter#generating-configuration) for details on how to generate a config file.
+
+## Blocks
+
+The following blocks are supported inside the definition of
+`prometheus.exporter.snmp` to configure collector-specific options:
+
+Hierarchy | Name | Description | Required
+--------- | ---- | ----------- | --------
+target | [target][] | Configures a SNMP target. | yes
+walk_param | [walk_param][] | SNMP connection profiles that can be used to override default SNMP settings. | no
+walk_param > auth | [auth][] | Configure auth for authenticating to the endpoint. | no
+
+[target]: #target-block
+[walk_param]: #walk_param-block
+[auth]: #auth-block
+
+### target block
```

Comment:
Each of these blocks should have a short intro sentence/paragraph like:

```markdown
The `target` block defines an individual SNMP target. The `target` block may be specified multiple times to define multiple targets. 
```

Ditto for walk_param and auth. See #3145 for specifics.